vllm/v1/worker/gpu_model_runner.py

```diff
@@ -1599,8 +1595,15 @@ def execute_model(
             if not broadcast_pp_output:
                 hidden_states.kv_connector_output = kv_connector_output
                 return hidden_states
-            get_pp_group().send_tensor_dict(hidden_states.tensors,
-                                            all_gather_group=get_tp_group())
+
+            all_gather_tensors = {
+                "residual":
+                not is_residual_scattered(self.vllm_config, num_input_tokens)
+            }
```

Comment:
Wait actually I'm confused. Aren't the hidden states distributed the same as the residuals? So if we use the allgather optimization on the residuals we should do the same for the hidden states?

Reply:
+1. That's my expectation as well. However it seems like a holdover from previous code: 

https://github.com/vllm-project/vllm/pull/24021/files#diff-80ee7e2a62f9dcfbb8a312dc4e3948557e97ef187290daebbcae1e28596bda29L1364-L1375

Reply:
Would be hard to tell that anything was wrong since perf would have improved, but not by as much as it should have I guess?